packages/atomic/src/components/common/atomic-focus-trap.tsx

```diff
@@ -0,0 +1,112 @@
+import {Component, Element, Listen, Prop, Watch} from '@stencil/core';
+
+function getParent(element: Element | ShadowRoot) {
+  if (element.parentNode) {
+    return element.parentNode as Element | ShadowRoot;
+  }
+  if (element instanceof ShadowRoot) {
+    return element.host;
+  }
+  return null;
+}
+
+function contains(
```

Comment:
These two utils could be extracted to a separate file, if you prefer.

Reply:
It's better to keep it internal if it's not used anywhere else.

And separate/extract only when it's needed.